dotcom-rendering/src/model/enhance-switches.ts

```diff
@@ -0,0 +1,21 @@
+export type ABTestSwitches = { [key: `ab${string}`]: boolean };
+
+/**
+ * Switches that start with "ab" are used for AB tests.
+ *
+ * We strip those out of the list of switches to reduce
+ * the amount of prop data that needs to be serialised
+ * for the SetABTests Island.
+ *
+ * Ref: [`frontend`][] Switches & [`ab-testing`][] filter.
+ *
+ * [`frontend`]: https://github.com/guardian/frontend/blob/016e9e26/common/app/conf/switches/ABTestSwitches.scala
+ * [`ab-testing`]: https://github.com/guardian/ab-testing/blob/b5de0e09/packages/ab-core/src/core.ts#L29-L30
+ *
+ * @param {Switches} switches
+ * @returns {ABTestSwitches} an object containing only AB switches
+ */
+export const filterABTestSwitches = (switches: Switches): ABTestSwitches =>
+	Object.fromEntries(
+		Object.entries(switches).filter(([key]) => key.startsWith('ab')),
```

Comment:
should we downcase the key string before matching on it? in case someone writes it as 'AB' instead of 'ab'?

Reply:
Abysmal, abhorrent, abominable! 😝 

From [the frontend docs](https://github.com/guardian/frontend/blob/37efbc8d62a118f9842a64eb17909adbcb55d8e7/docs/03-dev-howtos/01-ab-testing.md#adding-a-switch)
> The convention is that the test id has to start with the characters 'ab-'.

There might be other switches that start with these characters, but failing to filter them out is not a major issue. There’s just a few bytes more data sent over the wire.